### ISSUE TITLE
Conversation Broken with messages out of sync (formerly Corrupted Messages)

### ISSUE THREAD

**Issue by xanderdunn** [signalapp/Signal-iOS#1415]
### Bug description

A contact will randomly show an immense number of "Received corrupted message" messages.  
### Steps to reproduce
- Send a message or receive a message or simply open the app

**Actual result:** Shit ton of corrupted messages reports.  All future messages sent will be received as corrupted until the session is reset.
**Expected result:** No corrupted messages.
### Screenshots

![bug](https://cloud.githubusercontent.com/assets/1313618/19662532/c474989c-99ec-11e6-82e1-c8807fe55733.jpg)
### Device info

**Device**: iPhone 7
**iOS version**: 10.0.3.  The other user in my chat is on iOS 10.1 beta.  This could be related.
**Signal version:** 2.6.2
### Link to debug log

https://gist.github.com/anonymous/786ca6df4fce583019c5e0c661efba24

**Comment by perra22** [comment]
Hey, same issue here. Also tried to delete and reinstall the app and when I insert the verification code it doesn't work and it freezes.

Device: iPhone 7
iOS version: 10.1
Signal version: 2.6.2

**Comment by michaelkirk** [comment]
@perra22 - While registering you might be experiencing something like https://github.com/WhisperSystems/Signal-iOS/issues/1329. Until it's fixed, make sure you have "background refresh" enabled when registering for Signal.

More registration help:
http://support.whispersystems.org/hc/en-us/articles/213132457-Are-you-having-trouble-verifying-on-Signal-

**Comment by perra22** [comment]
Thank you @michaelkirk

**Comment by xanderdunn** [comment]
The sudden corrupted messages problem has occurred for me three times with this particular contact in the past couple of days.  

Fortunately, it's easy to remedy.  Resetting the session works.

**Comment by michaelkirk** [comment]
This is still an issue.

**Comment by michaelkirk** [comment]
@xanderdunn is this occurring after bringing your iOS device online after having a conversation via desktop? 

In your log it seems like a bunch of messages are getting synced all at once (at 2016/10/24 13:12)

If that's the case, do you know how long your phone was offline for?

**Comment by xanderdunn** [comment]
@michaelkirk: Thanks for looking at this.  For what it's worth, I haven't noticed this bug since the original three times I mentioned above.  I'm currently on 2.6.5.  

When the bug occurred, it did seem to attempt receiving a lot of messages.  However, there wasn't a clear reason for it.  My phone wasn't offline any of the three times it happened.  I was actually actively messaging in the iOS app all three times it occurred.  I would be successfully sending and receiving messages and then it would fail abruptly.

**Comment by michaelkirk** [comment]
That’s interesting. In case we can rule it out, were you linked to a desktop instance at all?

**Comment by xanderdunn** [comment]
@michaelkirk: Yes, both I and the person I was talking to were linked to a single desktop account.

**Comment by riyapenn** [comment]
Most recently provided log: https://gist.github.com/0c484853502b8d8823b3228f8393ccde 
"This happened to an ongoing chat with someone in another country while I had a (non-signal) phone conversation. So my data connection was WiFi while I used signal for chat. During my chat I setup a regular phonecall and the error messages appeared. "

**Comment by riyapenn** [comment]
Log URL: https://gist.github.com/ba55ec135b012702b7b18078012de4a4 
Tell us about the issue: 
I keep getting these messages when someone sends me a message: 
"Received message was out of sync. Tap to reset your secure session." 
"No available session for contact." 
No luck with updating the app. And the other person I'm texting via signal is getting my messages just fine.

**Comment by riyapenn** [comment]
**iOS log** https://gist.github.com/afc240616e77f46f01533017bf0d7838 

**Steps**
1. Link 2nd Signal Desktop to iOS
2. Receive message from unsaved number in a group thread at 1:57PM
3. See "Received message was out of sync. Tap to reset your secure session." on Signal iOS (no alert on Signal Desktop)
4. Reset session on Signal iOS
5. Do not see message in group thread on Signal iOS
6. See message in group thread on Signal Desktop

<img width="198" alt="screen shot 2016-12-20 at 2 04 06 pm" src="https://cloud.githubusercontent.com/assets/4898366/21369847/bd25e2f0-c6bd-11e6-8fb5-4115e7f9bd74.png">

**Comment by riyapenn** [comment]
Another user log: https://gist.github.com/99cc7d5f8f8dba1b07ff2f97b564768d 
No details on steps/scenario

Will add any new logs to this comment in the future.

**Comment by qayshp** [comment]
I just had this problem, and as suggested above by @michaelkirk , I was using a synced desktop, and the messages were readable on there.

**Comment by riyapenn** [comment]
New logs:
User 1: Signal iOS v 2.6.14 -- https://gist.github.com/2525dd027595f6ce6153d4fa5aca4f6d 
<img width="215" alt="screen shot 2017-01-31 at 2 23 36 pm" src="https://cloud.githubusercontent.com/assets/4898366/22487088/e4ef5efa-e7c0-11e6-8b73-c74adf8b1a1e.png">
(Display issued in https://github.com/WhisperSystems/Signal-iOS/issues/1677)

**Comment by michaelkirk** [comment]
https://github.com/WhisperSystems/Signal-iOS/issues/1677 seems to be a separate issue. I'll post some notes there.

**Comment by caveman1973** [comment]
Happened to me twice in the last few days to the same contact I was talking to. 

Could be a coincidence but I was using a VPN and I was in an area were the connection jumped from LTE to Wi-Fi, causing internet connection instability.

I don't use Signal desktop.


Is it safe to reset the session?

**Comment by michaelkirk** [comment]
@caveman1973 - do you know what version you were running when you last saw this? Could you submit a debug log now? And again if you ever see this behavior (preferably immediately after it happens) so we can track it down?

**Comment by riyapenn** [comment]
Recent log experiencing "No available session for contact.": https://gist.github.com/281b487b35b9fa49df836b727680ce3e

**Comment by riyapenn** [comment]
Log from iOS 2.11.4.1 which is the receiver of message seeing this alert ~3:52PM: https://gist.github.com/e5ea38d2d756a40d8bd7e918c6df8176

Sender (on Android) log: https://gist.github.com/968b359cf2c4f2bf638b0f11e3086100
<img width="237" alt="screen shot 2017-05-09 at 4 03 26 pm" src="https://cloud.githubusercontent.com/assets/4898366/25876308/1d370850-34d1-11e7-840f-66cdc7db596e.png">

**Comment by michaelkirk** [comment]
> Log from iOS 2.11.4.1 which is the receiver of message seeing this alert ~3:52PM: https://gist.github.com/e5ea38d2d756a40d8bd7e918c6df8176
>
> Sender (on Android) log: https://gist.github.com/968b359cf2c4f2bf638b0f11e3086100

@riyapenn can you lay out the user scenario a bit? Maybe a screenshot from android from the same time would clarify.

Main questions that come to mind:

It looks like Android re-registered before 3:52 when iOS received an out of sync message.

What action was Android taking? Sending a message? Making a call?

Was this the first message sent from Android->iOS since Android re-registered?

Had you recently done a "reset session" on a desktop linked with the Android?

**Comment by zjalexander** [comment]
I am getting this consistently between one of my contacts and me. 
We primarily use the iOS app to talk to each other, no problem.

Within the past 2 days, whenever one of us opens the desktop app (after an admittedly long time of not having the desktop app open) the other immediately gets about 30 "Received message was out of sync" messages.
Desktop log immediately after opening it and my contact got 30 messages: https://debuglogs.org/b769f61ad74762ac2b762a238379c2893ff09f172560236c37705974c6356478
My iOS log https://debuglogs.org/b8df08b9269f31c03f0d1cb75e03f484b478485f50e53554d9959477def072ce.zip

I will try to get my contact to upload their logs as well.

**Comment by michaelkirk-signal** [comment]
Hey @zjalexander - this is a really old Github issue. Over the years several completely different causes have resulted in this generic symptom - all the prior ones have been addressed.

Piling on a new bug report to the existing subscribers of this issue isn't great, but that's our fault for not closing this issue, not your fault for searching for similar symptoms (thanks for that actually, usually it's super helpful!).

In the interest of the above, and keeping your problem organized, I've moved your report to a new issue and will be closing this one.

See you over at https://github.com/signalapp/Signal-iOS/issues/4075
